editor/src/components/navigator/navigator-item/navigator-item.spec.browser2.tsx

```diff
@@ -0,0 +1,329 @@
+          true ? (
+            <div
+              style={{
+                backgroundColor: '#aaaaaa33',
+                display: 'grid',
+                position: 'absolute',
+                left: 70,
+                top: 10,
+                width: 50,
+                height: 50,
+              }}
+              data-uid='bbb'
+              data-testid='bbb'
+            />
+          ) : null
+        }
+        {
+          <div
+            style={{
+              backgroundColor: '#aaaaaa33',
+              position: 'absolute',
+              left: 130,
+              top: 10,
+              width: 50,
+              height: 50,
+              display: 'flex',
+            }}
+            data-uid='ccc'
+            data-testid='ccc'
+          />
+        }
+        {[0].map(() => (
+          <div
+            style={{
+              backgroundColor: '#aaaaaa33',
+              position: 'absolute',
+              left: 190,
+              top: 10,
+              width: 50,
+              height: 50,
+              display: 'flex',
+              flexDirection: 'column',
+            }}
+            data-uid='ddd'
+            data-testid='ddd'
+          />
+        ))}
+        <React.Fragment />
+        <div data-uid='72d'>
+          <div
+            style={{
+              backgroundColor: '#aaaaaa33',
+              position: 'absolute',
+              left: 215,
+              top: 136,
+              width: 90,
+              height: 68,
+            }}
+            data-uid='b4c'
+          >
+            {'text'}
+          </div>
+        </div>
+        <Scene
+          style={{
+            backgroundColor: '#aaaaaa33',
+            position: 'absolute',
+            left: -37,
+            top: 474,
+            width: 116,
+            height: 86,
+          }}
+          data-uid='a9c'
+        />
+        <img
+          src='https://github.com/concrete-utopia/utopia/blob/master/editor/resources/editor/pyramid_fullsize@2x.jpg?raw=true'
+          alt='Utopia logo'
+          style={{ height: 100, width: 100 }}
+          data-uid='922'
+        />
+      </Storyboard>
+    )
+  `
+
+  it('Should show the correct icons for each type of row', async () => {
+    const editor = await renderTestEditorWithCode(testProjectCode, 'await-first-dom-report')
+    const { jsxMetadata, elementPathTree, allElementProps } = editor.getEditorState().editor
+    const visibleNavigatorTargets = editor.getEditorState().derived.visibleNavigatorTargets
+
+    async function checkNavigatorIcon(
+      description: string,
+      expectedIcnProps: { category: string; type: string } | 'no-icon',
+      navigatorEntry: NavigatorEntry,
+      expectWarningIcon: 'expect-warning-icon' | 'expect-normal-icon' = 'expect-normal-icon',
+    ) {
+      const testId = layoutIconTestIdForEntry(navigatorEntry)
+
+      if (expectedIcnProps === 'no-icon') {
+        await expect(async () => editor.renderedDOM.getByTestId(testId)).rejects.toThrow()
```

Comment:
Nit: according to the [cheat sheet](https://testing-library.com/docs/dom-testing-library/cheatsheet/), you could use `queryByTestId` and expect it returns a null. furthermore, queryByTestId synchronously returns null, so you can remove the extra await from here